src/core/meshcop/joiner.cpp

```diff
@@ -144,6 +144,14 @@ otError Joiner::Start(const char *     aPskd,
 
     VerifyOrExit(mState == OT_JOINER_STATE_IDLE, error = OT_ERROR_BUSY);
 
+    if (!Get<ThreadNetif>().IsUp())
+    {
+        Get<ThreadNetif>().Up();
+    }
+
+    // Verify Thread stack is not fully enabled.
+    VerifyOrExit(Get<Mle::Mle>().GetRole() == Mle::kRoleDisabled, error = OT_ERROR_INVALID_STATE);
+
```

Comment:
```suggestion

    // Verify Thread stack is not fully enabled.
    VerifyOrExit(Get<Mle::Mle>().GetRole() == Mle::kRoleDisabled, error = OT_ERROR_INVALID_STATE);

    if (!Get<ThreadNetif>().IsUp())
    {
        Get<ThreadNetif>().Up();
    }

```
nit: verify before doing any operation?